src/wallet/partialTxProposal.ts

```diff
@@ -48,44 +60,90 @@ class PartialTxProposal {
    *
    * @returns {PartialTxProposal}
    */
-  static fromPartialTx(serialized: string, network: Network) {
+  static fromPartialTx(serialized: string, network: Network): PartialTxProposal {
     const partialTx = PartialTx.deserialize(serialized, network);
     const proposal = new PartialTxProposal(network);
     proposal.partialTx = partialTx;
     return proposal;
   }
 
+  /**
+   * Get all available utxos on the wallet history for a token
+   * and enrich them with tokenData.
+   *
+   * @param {HathorWallet} wallet Wallet which will provide the tokens.
+   * @param {string?} [token='00'] UID of token that is being sent
+   *
+   * @returns {UtxoExtended[]}
+   */
+  static getWalletUtxos(
+    wallet: HathorWallet,
+    token: string = HATHOR_TOKEN_CONFIG.uid,
+  ): UtxoExtended[] {
+    const historyTransactions = wallet.getFullHistory();
+    const allUtxos = [...wallet.getAllUtxos({ token })].filter(utxo => utxo.authorities === 0);
```

Comment:
Maybe, I think it would be a good utility method but I think we should consider this on another issue.
We have other utxos methods and we may want to unify/rework them to make it easier to use and understand.